src/utils.tsx

```diff
@@ -152,3 +153,28 @@ export function isOptionFacet(
 ): facet is PlpMultipleFacet | PlpSingleFacet | PlpHierarchicalFacet {
   return isMultipleOrBucketedFacet(facet) || isSingleFacet(facet) || isHierarchicalFacet(facet);
 }
+
+export function getSearchCnstrcDataAttributes(
+  data: Nullable<PlpSearchData>,
+  initialResponse: SearchResponse | undefined,
+) {
+  let totalNumResults: number | undefined;
+  let dataCnstrc: Record<`data-cnstrc-${string}`, any> = { 'data-cnstrc-search': true };
```

Comment:
I think we should ensure that data: `Nullable<PlpSearchData>` is not null before assigning it as `data-cnstrc-search`